src/assets/TokensController.test.ts

```diff
@@ -312,4 +293,458 @@ describe('TokensController', () => {
+        'Must specify address, symbol, and decimals.',
+      );
+    });
+
+    it('should error if symbol is not defined', async function () {
+      asset.symbol = undefined;
+      const result = tokensController.watchAsset(asset, type);
+      await expect(result).rejects.toThrow(
+        'Must specify address, symbol, and decimals.',
+      );
+    });
+
+    it('should error if symbol is empty', async function () {
+      asset.symbol = '';
+      const result = tokensController.watchAsset(asset, type);
+      await expect(result).rejects.toThrow(
+        'Must specify address, symbol, and decimals.',
+      );
+    });
+
+    it('should error if symbol is too long', async function () {
+      asset.symbol = 'ABCDEFGHIJKLM';
+      const result = tokensController.watchAsset(asset, type);
+      await expect(result).rejects.toThrow(
+        'Invalid symbol "ABCDEFGHIJKLM": longer than 11 characters.',
+      );
+    });
+
+    it('should error if decimals is invalid', async function () {
+      asset.decimals = -1;
+      const result = tokensController.watchAsset(asset, type);
+      await expect(result).rejects.toThrow(
+        'Invalid decimals "-1": must be 0 <= 36.',
+      );
+
+      asset.decimals = 37;
+      const result2 = tokensController.watchAsset(asset, type);
+      await expect(result2).rejects.toThrow(
+        'Invalid decimals "37": must be 0 <= 36.',
+      );
+    });
+
+    it('should error if address is invalid', async function () {
+      asset.address = '0x123';
+      const result = tokensController.watchAsset(asset, type);
+      await expect(result).rejects.toThrow('Invalid address "0x123".');
+    });
+
+    it('should handle ERC20 type and add to suggestedAssets', async function () {
+      const clock = sinon.useFakeTimers(1);
+      sinon
+        .stub(tokensController, '_generateRandomId')
+        .callsFake(() => '12345');
+      type = 'ERC20';
+      await tokensController.watchAsset(asset, type);
+      await expect(tokensController.state.suggestedAssets).toStrictEqual([
+        {
+          id: '12345',
+          status: 'pending',
+          time: 1, // uses the fakeTimers clock
+          type: 'ERC20',
+          asset,
+        },
+      ]);
+      clock.restore();
+    });
+
+    it('should add token correctly if user confirms', async function () {
+      sinon
+        .stub(tokensController, '_generateRandomId')
+        .callsFake(() => '12345');
+      type = 'ERC20';
+      await tokensController.watchAsset(asset, type);
+
+      await tokensController.acceptWatchAsset('12345');
+      await expect(tokensController.state.suggestedAssets).toStrictEqual([]);
+      await expect(tokensController.state.tokens).toHaveLength(1);
+      await expect(tokensController.state.tokens).toStrictEqual([
+        {
+          isERC721: false,
+          ...asset,
+        },
+      ]);
+    });
+    it('should add a valid suggested asset via watchAsset', async () => {
```

Comment:
This seems like a duplicate of `should handle ERC20 type and add to suggestedAssets` :thinking:

Reply:
Good call! Removing